packages/ember/index.js

```diff
@@ -651,8 +651,27 @@ if (ENV.EXTEND_PROTOTYPES.String) {
     return htmlSafe(this);
   };
 }
-Ember.String.htmlSafe = htmlSafe;
-Ember.String.isHTMLSafe = isHTMLSafe;
+const deprecateImportFromString = function (
+  name,
+  fn,
+  message = `Importing ${name} from '@ember/string' is deprecated. Please import ${name} from '@ember/template' instead.`
+) {
+  return function () {
+    deprecate(message, false, {
+      id: 'ember-string.htmlsafe-ishtmlsafe',
+      for: 'ember-source',
+      since: {
+        available: '3.24',
+      },
+      until: '4.0.0',
+      url: 'https://deprecations.emberjs.com/v3.x/#toc_ember-string-htmlsafe-ishtmlsafe',
+    });
+
+    return fn(...arguments);
+  };
+};
+Ember.String.htmlSafe = deprecateImportFromString('htmlSafe', htmlSafe);
+Ember.String.isHTMLSafe = deprecateImportFromString('isHTMLSafe', isHTMLSafe);
```

Comment:
Instead of returning a function that does the deprecation, can you make these a getter that issues the deprecation upon access (those getters would then return the original values)? There are other examples like that in this file.

Reply:
That would look something like:

```js
Object.defineProperty(Ember.String, 'htmlSafe', {
  enumerable: true,
  configurable: true,
  get() {
    // do the deprecation
    // then return the normal value
  }
});
```

Reply:
Should `htmlSafe` and `isHTMLSafe` get separate deprecation ids? I just made up `ember-string.htmlsafe-ishtmlsafe`.  Is there a pattern I should be following here?

Reply:
I also just invented the URL https://deprecations.emberjs.com/v3.x/#toc_ember-string-htmlsafe-ishtmlsafe
I'm not sure if there's a pattern I should follow there as well (I was looking at #19234).

Reply:
> Should `htmlSafe` and `isHTMLSafe` get separate deprecation ids? I just made up `ember-string.htmlsafe-ishtmlsafe`.  Is there a pattern I should be following here?

Dealers choice (I think either is fine)😊

Reply:
> I also just invented the URL https://deprecations.emberjs.com/v3.x/#toc_ember-string-htmlsafe-ishtmlsafe
> 
> I'm not sure if there's a pattern I should follow there as well (I was looking at #19234).

We usually figure out the urls by making the PR and grabbing them from the running app. Once the deprecation guide PR is merged, we can ensure that the URL is correct, then we can land this.

Reply:
Deprecation app PR: https://github.com/ember-learn/deprecation-app/pull/758